src/lavinmq/federation/upstream.cr

```diff
@@ -63,7 +63,10 @@ module LavinMQ
         if link = @ex_links[federated_exchange.name]?
           return link
         end
-        upstream_exchange = @exchange ||= federated_exchange.name
+        upstream_exchange = @exchange || federated_exchange.name
+        if upstream_exchange.nil? || upstream_exchange.empty?
```

Comment:
if we get true for `.nil?` here, we will reassign it `nil`? since if `@exchange` is `nil`, we will have assigned `federated_exchange.name `
Do we really need to do the check for `upstream_exchange.nil?` then?